src/chart/helper/createClipPathFromCoordSys.ts

```diff
@@ -43,7 +44,14 @@ function createGridClipPath(
     let width = rect.width;
     let height = rect.height;
 
-    const lineWidth = seriesModel.get(['lineStyle', 'width']) || 2;
+    const zr = cartesian.model.ecModel.scheduler.ecInstance.getZr();
+    const dpr = zr.painter instanceof CanvasPainter
```

Comment:
A better way to get dpr is from https://github.com/apache/echarts/blob/master/src/core/echarts.ts#L703 . You may add a new parameter to this function like `dpr`.

Reply:
Oh yes, it's better, I'll change it soon.